lib/resource-serializer.js

```diff
@@ -1,9 +1,22 @@
 'use strict';
+var _ = require('lodash');
 var SerializerUtils = require('./serializer-utils');
 
 function ResourceSerializer(collectionName, record, opts) {
   var payload = { data: {} };
 
+  function getLinks(links) {
+    return _.mapValues(links, function (value) {
+      if (_.isFunction(value)) {
+        return value(records);
+      } else {
+        return value;
+      }
+    });
+  }
+
+  if (opts.topLevelLinks) { payload.links = getLinks(opts.topLevelLinks); }
+
```

Comment:
Exactly, I think the `serializer-utils` is the best place. Do you want some help or do you prefer to propose your own solution?